src/NServiceBus.Core/ObjectBuilder/Common/CommonObjectBuilder.cs

```diff
@@ -3,101 +3,70 @@ namespace NServiceBus
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using System.Threading;
     using ObjectBuilder;
     using ObjectBuilder.Common;
 
-    class CommonObjectBuilder : IBuilder, IConfigureComponents
+    sealed class CommonObjectBuilder : IBuilder, IConfigureComponents
     {
-        public CommonObjectBuilder(IContainer container)
+        readonly AsyncLocal<IBuilder> currentBuilder;
+        readonly IBuilder previous;
```

Comment:
nitpick: should this be named `parent` given we also use "child" as the terminology for nested scopes?

Reply:
Fixed and also moved the fields